pkg/pod/pod.go

```diff
@@ -336,3 +342,22 @@ func shouldOverrideWorkingDir(kubeclient kubernetes.Interface) bool {
 	}
 	return true
 }
+
+// shouldAddReadyAnnotationonPodCreate returns a bool indicating whether the
+// controller should add the `Ready` annotation when creating the Pod. The
+// default behavior is to add the Ready annotation after  Pod has been created
+// and all of its sidecars are in a Ready state.
```

Comment:
totally minor point but I find when I find myself writing docstrings and tests for a private/unexported function, it's often an indication that it would make senes to move that function into a separate/new function and make it exported

Reply:
Agreed. I think pkg/pod/pod.go has a few of these feature flags functions now. We have #2363 open for making this better!